File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        merged_arr = nums1+nums2
        merged_arr.sort()
        ln = len(merged_arr)
        if ln%2==0:
            sum = merged_arr[ln//2-1]+merged_arr[ln//2]
            return sum/2.0
        else:
            return merged_arr[ln//2]

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void* a, const void* b) {
    return (*(int*)a - *(int*)b);
}

float findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int* merged = (int*)malloc((nums1Size + nums2Size) * sizeof(int));
    memcpy(merged, nums1, nums1Size * sizeof(int));
    memcpy(&merged[nums1Size], nums2, nums2Size * sizeof(int));

    qsort(merged,(nums1Size + nums2Size),sizeof(int),compare);

    int length = nums1Size + nums2Size;
    if (length % 2 == 0) {
        return ((float)merged[length / 2 - 1] + (float)merged[length / 2]) / 2.0;
    } else {
        return (float)merged[length / 2];
    }

    free(merged);
}
